src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs

```diff
@@ -737,6 +735,29 @@ internal ArrowOrDataFrameGroupedMapCommandExecutor(Version version)
             return ExecuteArrowGroupedMapCommand(inputStream, outputStream, commands);
         }
 
+        private RecordBatch WrapColumnsInStructIfApplicable(RecordBatch batch)
+        {
+            if (_version >= new Version(Versions.V3_0_0))
```

Comment:
maybe this instead ?

```suggestion
            if (_version.Major >= 3)
```

Reply:
Ha, that's actually what I had first, until I saw this:  https://github.com/dotnet/spark/blob/1bcd37b501eb08df72259e2770da6cf23e8b946a/src/csharp/Microsoft.Spark.E2ETest/Utils/VersionBasedFacts.cs#L14. 

Are you worried about the `new` ? I don't mind changing it!!

Reply:
just a very micro optimization.  We can keep, I don't mind either.  👍